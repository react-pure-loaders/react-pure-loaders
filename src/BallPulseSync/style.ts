import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';
import delay from '../utilities';

const amount = '10px';

const ballPulseSync = keyframes({
  '100%': {
    transform: 'translateY(0)',
  },
  '33%': {
    transform: `translateY(${amount})`,
  },
  '66%': {
    transform: `translateY(-${amount})`,
  },
});

interface Pulse {
    [id: string]: object;
}

export function ballPulse() {
  const n = 3;
  let i = 1;
  const grid: Pulse = {};
  for (; i < 4; i += 1) {
    grid[`> div:nth-of-type(${i})`] = {
      animation: `${ballPulseSync} 0.6s ${delay('0.07s', n, i)} infinite ease-in-out`,
    };
  }

  return grid;
}

export default (color?: string) => ({
  ...ballPulse(),
  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    display: 'inline-block',
  },
});
