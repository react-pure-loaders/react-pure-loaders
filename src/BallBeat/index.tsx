/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
