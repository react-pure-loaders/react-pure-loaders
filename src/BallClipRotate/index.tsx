/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
