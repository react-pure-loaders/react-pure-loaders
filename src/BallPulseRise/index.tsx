/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
