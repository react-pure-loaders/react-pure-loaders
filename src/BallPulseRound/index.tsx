/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
