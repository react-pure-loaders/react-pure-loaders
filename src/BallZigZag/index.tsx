/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
