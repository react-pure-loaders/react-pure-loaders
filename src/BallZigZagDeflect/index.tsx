/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
