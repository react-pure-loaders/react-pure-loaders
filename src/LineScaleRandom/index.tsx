/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
