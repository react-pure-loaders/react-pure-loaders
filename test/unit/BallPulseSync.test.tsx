import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallPulseSync from '../../src/BallPulseSync';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallPulseSync>', () => {
    afterEach(cleanup);

    test('BallPulseSync should match snapshot', () => {
        const { container } = render(<BallPulseSync loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('BallPulseSync should have default color', () => {
        const { container } = render(<BallPulseSync loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test('BallPulseSync should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallPulseSync color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('BallPulseSync should have no children', () => {
        const { container } = render(<BallPulseSync loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallPulseSync should have three children', () => {
        const { container } = render(<BallPulseSync loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(4);
    });
});
