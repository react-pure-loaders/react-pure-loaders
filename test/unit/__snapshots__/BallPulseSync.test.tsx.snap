// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallPulseSync> BallPulseSync should match snapshot 1`] = `
@keyframes animation-0 {
  100% {
    -webkit-transform: translateY(0);
    -moz-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }

  33% {
    -webkit-transform: translateY(10px);
    -moz-transform: translateY(10px);
    -ms-transform: translateY(10px);
    transform: translateY(10px);
  }

  66% {
    -webkit-transform: translateY(-10px);
    -moz-transform: translateY(-10px);
    -ms-transform: translateY(-10px);
    transform: translateY(-10px);
  }
}

@keyframes animation-0 {
  100% {
    -webkit-transform: translateY(0);
    -moz-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }

  33% {
    -webkit-transform: translateY(10px);
    -moz-transform: translateY(10px);
    -ms-transform: translateY(10px);
    transform: translateY(10px);
  }

  66% {
    -webkit-transform: translateY(-10px);
    -moz-transform: translateY(-10px);
    -ms-transform: translateY(-10px);
    transform: translateY(-10px);
  }
}

@keyframes animation-0 {
  100% {
    -webkit-transform: translateY(0);
    -moz-transform: translateY(0);
    -ms-transform: translateY(0);
    transform: translateY(0);
  }

  33% {
    -webkit-transform: translateY(10px);
    -moz-transform: translateY(10px);
    -ms-transform: translateY(10px);
    transform: translateY(10px);
  }

  66% {
    -webkit-transform: translateY(-10px);
    -moz-transform: translateY(-10px);
    -ms-transform: translateY(-10px);
    transform: translateY(-10px);
  }
}

.emotion-0>div:nth-of-type(1) {
  -webkit-animation: animation-0 0.6s -0.14s infinite ease-in-out;
  animation: animation-0 0.6s -0.14s infinite ease-in-out;
}

.emotion-0>div:nth-of-type(2) {
  -webkit-animation: animation-0 0.6s -0.07s infinite ease-in-out;
  animation: animation-0 0.6s -0.07s infinite ease-in-out;
}

.emotion-0>div:nth-of-type(3) {
  -webkit-animation: animation-0 0.6s 0.00s infinite ease-in-out;
  animation: animation-0 0.6s 0.00s infinite ease-in-out;
}

.emotion-0>div {
  border-radius: 100%;
  height: 15px;
  margin: 2px;
  width: 15px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  display: inline-block;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
</div>
`;
